packages/contracts/contracts/governance/GovernanceToken.sol

```diff
@@ -1,47 +1,113 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.15;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
-
-/// @custom:predeploy 0x4200000000000000000000000000000000000010
-/// @title GovernanceToken
-/// @notice The Kroma token used in governance and supporting voting and delegation. Implements
-///         EIP 2612 allowing signed approvals. Contract is "owned" by a `MintManager` instance with
-///         permission to the `mint` function only, for the purposes of enforcing the token
-///         inflation schedule.
-contract GovernanceToken is ERC20Burnable, ERC20Votes, Ownable {
-    /// @notice Constructs the GovernanceToken contract.
-    constructor() ERC20("Kroma", "KRO") ERC20Permit("Kroma") { }
-
-    /// @notice Allows the owner to mint tokens.
-    /// @param _account The account receiving minted tokens.
-    /// @param _amount  The amount of tokens to mint.
-    function mint(address _account, uint256 _amount) public onlyOwner {
+
+import { Predeploys } from "../libraries/Predeploys.sol";
+import { KromaMintableERC20 } from "../universal/KromaMintableERC20.sol";
+
+/**
+ * @custom:proxied
+ * @custom:predeploy 0x42000000000000000000000000000000000000ff
+ * @title GovernanceToken
+ * @notice The KRO token used in governance and supporting voting and delegation. Implements
+ *         EIP 2612 allowing signed approvals. It can be bridged to any other specified chain.
+ *         This token has a cap of 50m in total supply, and is minted once every block by the MintManager.
+ */
+contract GovernanceToken is KromaMintableERC20, ERC20Burnable, ERC20Votes {
```

Comment:
I added ERC20Votes for the following reasons.

1. In Optimism, the GovernanceToken token inherits from ERC20Votes.
2. `ValidatorManager` is the contract for **validator** activities. It should have features for token-based governance. This would be available in Tally, etc.

Reply:
I agree with the first point, and the point that the governance system should take the tokens inside `ValidatorManager` into account. The reason of my thoughts was that `ERC20Votes` is designed to call `_update` function when token transfer happens, and delegators should transfer the tokens to `ValidatorManager` to delegate, so that most of governance power will remain at `ValidatorManager` uselessly. 

 But I think it can be handled through overriding `transferFrom` function and dividing cases, and we can leave this point for further discussion.

Reply:
Oh I see. Then we'll need to implement the token-base governance contract in a different way.